content/docs/reference/buildpack-api.md

```diff
@@ -187,13 +186,120 @@ The schema is as follows:
 
         - **`version`** _(string, required)_\
           The version of the buildpack being referred to.
-    
+
         - **`optional`** _(boolean, optional, default: `false`)_\
           Whether or not this buildpack is optional during detection.
- 
+
 - **`metadata`** _(any, optional)_\
     Arbitrary data for buildpack.
- 
+
+## Build Plan
+The [Build Plan](https://github.com/buildpacks/spec/blob/master/buildpack.md#build-plan-toml) is a document the buildpacks can use to pass information between the [detect](#bindetect) and [build](#bindetect) phases. The build plan is passed (by the lifecycle) as a parameter to the `detect` and `build` binaries of the buildpack.
+* During the `detect` phase, the buildpack(s) may write something it `requires` or `provides` (or both) into the Build Plan.
+* During the `build` phase, the buildpack(s) may read the Buildpack Plan (a condensed version of the Build Plan, composed by the lifecycle) to determine what it should do, and refine the Buildpack Plan with more exact metadata (eg: what version dependency it requires).
+
+A buildpack can `require` or `provide` multiple dependencies, and even multiple groupings of dependencies (using `or` lists). Additionally, multiple buildpacks may `require` or `provide` the same dependency.
+
+The lifecycle uses the `Build Plan` as one element in deciding whether or not a particular list of buildpacks is appropriate, by seeing whether all dependencies required can be provided by that list. See the [spec](https://github.com/buildpacks/spec/blob/master/buildpack.md#phase-1-detection) for particulars on how ordering buildpacks can adjust detection results.
+
+### Example
+Let's walk through some possible cases a `node-engine` buildpack may consider:
+
+1.  Nothing in the app explicitly calls out that it is needed
+2.  It is explicitly referred to in some configuration file
+3.  Two different configuration files request different versions of `node`
+
+We will also consider what a `NPM` buildpack may do.
+
+#### 1. No Explicit Request
```

Comment:
> If a required buildpack provides a dependency that is not required by the same buildpack or a subsequent buildpack, the trial MUST fail to detect.

Based on the spec, this would fail detection. Maybe we should side-note it.

Reply:
Makes sense!

Reply:
@jromero I added a note both by the `node` and `npm` cnb clarifications that running either alone wouldn't work, per the lifecycle